packages/form-js-viewer/src/render/hooks/useLabelCorrelation.js

```diff
@@ -0,0 +1,17 @@
+import { useCallback, useMemo } from 'preact/hooks';
+import { isObject } from 'min-dash';
+
+/**
+ * This hook allows us to retrieve the label from a value in linear time by caching it in a map
+ * @param {Array} options
+ */
+export function useLabelCorrelation(options) {
+
+  // This allows us to retrieve the label from a value in linear time
+  const labelMap = useMemo(() => Object.assign({}, ...options.map((o) => ({ [_getValueHash(o.value)]:  o.label }))), [ options ]);
```

Comment:
Did you have any performance issues? It's not a good idea to abuse `useMemo`,`useCallback`, etc because of memory leaks

Reply:
Yeah, this is why it's here and it's always been that way. I've just brought this pattern into a hook because we use it a few places. Otherwise the complexity is n^2 and we want the components to deal with values of n > 1000